### PR TITLE
Update 1.36 to 1.37 upgrade logic 

### PR DESCRIPTION
**Pull request by bernardkwok**
Update #696 
Update #393
- Update upgrade logic to 1.37 and add in new test
- Add in LookGroup Element support.

### CODE REVIEW

File: resources/Materials/TestSuite/stdlib/upgrade/1_36_to_1_37.mtlx [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+   </invert>
+   <invert name="invert2" type="matrix44">
+      <input name="in" type="matrix44" value="1.0, 0.0, 0.0, 0.0,  0.0, 1.0, 0.0, 0.0,  0.0, 0.0, 1.0, 0.0,  0.0, 0.0, 0.0, 1.0"/>
+   </invert>
+
+   <!-- no change for non-matrix invert variants -->
+   <invert name="invert_no_change_float" type="float">
+      <input name="in" type="float" value="0.5"/>
+   </invert>
+   <invert name="invert_no_change_vec2" type="vector2" xpos="5.75172" ypos="4.74">
```
Comment (feldstj) [inline resources/Materials/TestSuite/stdlib/upgrade/1_36_to_1_37.mtlx]: Are you sure you want to be saving the xpos, ypos info? Seem to be doing it in a lot of places, figured I'd check.

Comment (bernardkwok) [reply resources/Materials/TestSuite/stdlib/upgrade/1_36_to_1_37.mtlx]: Due to copy and paste from existing test cases. I'll remove them.

File: source/MaterialXCore/Document.cpp [hunk 2]
```diff
@@ -631,14 +631,125 @@ void Document::upgradeVersion()
...
+                }
+                // "intest" input becomes "value2" input
+                InputPtr intest = node->getInput(INTEST_INPUT);
+                if (intest)
+                {
+                    intest->setName(VALUE2_INPUT);
+                }
+            }
+
+            // Change notes with category "tranform[vector|point|normal]",
```
Comment (feldstj) [inline source/MaterialXCore/Document.cpp]: Should _notes_ be _nodes_?

Comment (bernardkwok) [reply source/MaterialXCore/Document.cpp]: Yes :)

File: source/MaterialXTest/Document.cpp [hunk 3]
```diff
@@ -108,3 +111,50 @@ TEST_CASE("Document", "[document]")
...
+    {
+        auto convertItem = convertSet.find(node->getCategory());
+        if (convertItem != convertSet.end())
+        {
+            convertItem->second++;
+        }
+    }
+    for (auto convertItem : convertSet)
+    {
+        REQUIRE((convertItem.second > 0));
```
Comment (feldstj) [inline source/MaterialXTest/Document.cpp]: Shouldn't this be == 1? > 0 might still be a bug, no?

File: source/MaterialXTest/Look.cpp [hunk 4]
```diff
@@ -93,3 +93,33 @@ TEST_CASE("Look", "[look]")
...
+    REQUIRE(looksVec.size() == looksVec2.size());
+
+    REQUIRE(lookGroup->getActiveLook().empty());
+    lookGroup->setActiveLook("look1");
+    REQUIRE(lookGroup->getActiveLook() == "look1");
+
+    doc->removeLookGroup("lookgroup1");
+    lookGroups = doc->getLookGroups();
+    REQUIRE(lookGroups.size() == 0);
+}
```
Comment (feldstj) [inline source/MaterialXTest/Look.cpp]: Should you have a newline here?

File: source/MaterialXTest/Document.cpp (hunk 3 above)
Comment (bernardkwok) [reply source/MaterialXTest/Document.cpp]: No there are type variants for the node categories so there can be > 1.

File: source/MaterialXTest/Look.cpp (hunk 4 above)
Comment (bernardkwok) [reply source/MaterialXTest/Look.cpp]: I'll fix.

File: source/MaterialXTest/Document.cpp [hunk 5]
```diff
@@ -108,3 +111,50 @@ TEST_CASE("Document", "[document]")
...
+        { "combine2", 1}, 
+        { "combine3", 1},
+        { "combine4", 1}
+    };
+    for (mx::NodePtr node : doc2->getNodes())
+    {
+        auto convertItem = convertSet.find(node->getCategory());
+        if (convertItem != convertSet.end())
+        {
+            convertItem->second--;
```
Comment (bernardkwok) [inline source/MaterialXTest/Document.cpp]: Added a reverse count to check for explicit number of conversions per category.

**Pull request merged**